packages/front-end/components/Settings/EditDataSource/DataSourceExperimentProperties/DataSourceViewEditExperimentProperties.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, useCallback, useState } from "react";
+import { DataSourceQueryEditingModalBaseProps } from "../types";
+import { FaPencilAlt } from "react-icons/fa";
+import { DataSourceEditExperimentEventPropertiesModal } from "./DataSourceEditExperimentEventPropertiesModal";
+import {
+  DataSourceEvents,
+  DataSourceInterfaceWithParams,
+} from "back-end/types/datasource";
+import cloneDeep from "lodash/cloneDeep";
+
+type DataSourceViewEditExperimentPropertiesProps = DataSourceQueryEditingModalBaseProps;
+
+export const DataSourceViewEditExperimentProperties: FC<DataSourceViewEditExperimentPropertiesProps> = ({
+  onSave,
+  onCancel,
+  dataSource,
+}) => {
+  const [uiMode, setUiMode] = useState<"view" | "edit">("view");
+
+  const handleEdit = useCallback(() => {
+    setUiMode("edit");
+  }, []);
+
+  const handleCancel = useCallback(() => {
+    setUiMode("view");
+    onCancel();
+  }, [onCancel]);
+
+  const handleSave = useCallback((eventProperties: DataSourceEvents) => {
+    const copy = cloneDeep<DataSourceInterfaceWithParams>(dataSource);
+    copy.settings.events = eventProperties;
+    onSave(copy);
+  }, []);
+
+  if (!dataSource) {
+    console.error("ImplementationError: dataSource cannot be null");
+    return null;
+  }
+
+  const dataSourceEvents = dataSource.properties.events;
+  if (!dataSourceEvents) {
+    console.error(
+      "ImplementationError: dataSource.properties.events cannot be null"
+    );
+    return null;
+  }
```

Comment:
Similar comment to above, maybe we just always render this component and just put empty strings as the value.  That way, if they don't have anything set yet, for whatever reason, they will be able to get out of that broken state.